src/tests/Microsoft.PowerFx.Core.Tests/NamedFormulasTests.cs

```diff
@@ -69,5 +70,28 @@ public void GetNamedFormulasTest(string script, string expectedX, string expecte
             Assert.Equal(expectedX, formulas.ElementAt(0).formula.Script);
             Assert.Equal(expectedY, formulas.ElementAt(1).formula.Script);
         }
+
+        [Theory]
+        [InlineData("x = 1;x = 2;", "1", 1, "x")]
+        [InlineData("y = 1;y = 2;y = 2;", "1", 2, "y")]
+        [InlineData("y = 1;y = 2;y=3", "1", 3, "y")]
+        public void NamedFormulasWithDuplicateVariablesTest(string script, string expectedX, int expectedErrorCount, string errorToken)
+        {
+            var namedFormula = new NamedFormulas(script);
+            Assert.False(namedFormula.EnsureParsed());
+
+            var errors = namedFormula.GetParseErrors();
+            Assert.NotEmpty(errors);
+            Assert.Equal(errors.Count(), expectedErrorCount);
+
+            var identifierDName = errors.ElementAt(0)?.Tok.As<IdentToken>()?.Name;
+            Assert.NotNull(identifierDName);
+            Assert.Equal(errorToken, identifierDName.Value);
+
+            var formulas = namedFormula.GetNamedFormulas();
+            Assert.NotNull(formulas);
+
+            Assert.Equal(expectedX, formulas.ElementAt(0).formula.Script);
```

Comment:
>Assert.Equal(expectedX, formulas.ElementAt(0).formula.Script); [](http://example.com/codeflow?start=12&length=62)

what is this testing?

Reply:
I am checking with the first assigned value. 
for example: x=1;x=2; 
in this case the value of x should be 1.